File: exp-01/q1-tower-of-hanoi/solution.js
const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

let count = 0;

function towerOfHanoi(n, from, to, aux) {
    if (n === 1) {
        console.log(`Move disk 1 from ${from} to ${to}`);
        count++;
        return;
    }
    towerOfHanoi(n - 1, from, aux, to);
    console.log(`Move disk ${n} from ${from} to ${to}`);
    count++;
    towerOfHanoi(n - 1, aux, to, from);
}

rl.on('line', (line) => {
    const n = parseInt(line.trim(), 10);
    towerOfHanoi(n, 'A', 'C', 'B');
    console.log(`Total moves: ${count}`);
    rl.close();
});

File: exp-01/q1-tower-of-hanoi/solution.cpp
#include <iostream>
using namespace std;

void towerOfHanoi(int n, char from, char to, char aux, int &count)
{
    if (n == 1)
    {
        cout << "Move disk 1 from " << from << " to " << to << endl;
        count++;
        return;
    }
    towerOfHanoi(n - 1, from, aux, to, count);
    cout << "Move disk " << n << " from " << from << " to " << to << endl;
    count++;
    towerOfHanoi(n - 1, aux, to, from, count);
}

int main()
{
    int n, count = 0;
    cin >> n;
    towerOfHanoi(n, 'A', 'C', 'B', count);
    cout << "Total moves: " << count << endl;
    return 0;
}